launcher/minecraft/mod/tasks/LocalResourcePackParseTask.cpp

```diff
@@ -186,7 +349,21 @@ bool processMCMeta(ResourcePack& pack, QByteArray&& raw_data)
         auto pack_obj = Json::requireObject(json_doc.object(), "pack", {});
 
         pack.setPackFormat(Json::ensureInteger(pack_obj, "pack_format", 0));
-        pack.setDescription(Json::ensureString(pack_obj, "description", ""));
+
+        // description could either be string, or array of dictionaries
+        auto desc_val = pack_obj.value("description");
+
+        if (desc_val.isString()) {
+            pack.setDescription(desc_val.toString());
```

Comment:
The `desc_val` can also be any JSON type: 
```
A raw JSON text component can be any JSON data type other than JSON null, including JSON string, JSON array, JSON object, JSON boolean, and JSON number
```
so you would need to also handle null,object, boolean, and numbers.